integritybackend/actions.py

```diff
@@ -8,159 +8,283 @@
-        asset_helper = AssetHelper(org_id)
-        file_util = FileUtil()
-
-        collection = config.ORGANIZATION_CONFIG.get_collection(
-            org_id, collection_id
-        )
-        action = config.ORGANIZATION_CONFIG.get_action(
-            org_id, collection_id, "archive"
-        )
-        action_params = action.get('params')
-        if action_params["encryption"]["algo"] != "aes-256-cbc":
-            raise Exception(
-                f"Encryption algo {action_params['encryption']['algo']} not implemented"
-            )
-
-        # Verify ZIP name
-        input_zip_sha = os.path.splitext(os.path.basename(zip_path))[0]
-        if input_zip_sha != file_util.digest_sha256(zip_path):
-            raise Exception(
-                f"SHA-256 of ZIP does not match file name: {zip_path}"
-            )
+        try:
+            # TODO: change function to take just org_id as param
+            action_name = "archive"
+            org_id = org_config["id"]
+
+            asset_helper = AssetHelper(org_id)
+            _file_util = FileUtil()
+
+            collection = config.ORGANIZATION_CONFIG.get_collection(org_id, collection_id)
+            action = config.ORGANIZATION_CONFIG.get_action(org_id, collection_id, action_name)
+            action_params = action.get("params")
+            if action_params["encryption"]["algo"] != "aes-256-cbc":
+                raise Exception(
+                    f"Encryption algo {action_params['encryption']['algo']} not implemented"
+                )
 
-        # Copy ZIP
-        archive_dir = asset_helper.get_action_dir(collection_id, "archive")
-        tmp_zip = shutil.copy2(zip_path, archive_dir)
+            # Verify ZIP name
+            input_zip_sha = os.path.splitext(os.path.basename(zip_path))[0]
+            if input_zip_sha != _file_util.digest_sha256(zip_path):
+                raise Exception(f"SHA-256 of ZIP does not match file name: {zip_path}")
+
+            # Copy ZIP
+            archive_dir = asset_helper.path_for_action(collection_id, action_name)
+            tmp_zip = shutil.copy2(zip_path, archive_dir)
+
+            # Verify ZIP contents are valid and expected
+            zip_listing = zip_util.listing(tmp_zip)
+            if len(zip_listing) != 3:
+                # ZIP must contain three files: content, meta-content , meta-recorder
+                raise Exception(
+                    f"ZIP at {zip_path} has more than three files: {zip_listing}"
+                )
+            content_filename = next((s for s in zip_listing if "-meta-" not in s), None)
+            if content_filename is None:
+                raise Exception(f"ZIP at {zip_path} has no content file: {zip_listing}")
+            if "/" in content_filename:
+                raise Exception(f"Content file is not at ZIP root: {content_filename}")
+            if (
+                os.path.splitext(content_filename)[1][1:]
+                not in collection["asset_extensions"]
+            ):
```

Comment:
```suggestion
            if (
                '*' not in collection["asset_extensions"] and
                os.path.splitext(content_filename)[1][1:]
                not in collection["asset_extensions"]
            ):
```

Add support for any filetype inside zip